sympy/core/relational.py

```diff
@@ -434,7 +471,11 @@ def __new__(cls, lhs, rhs=None, **options):
                 if L is False:
                     if lhs == -rhs:  # Eq(oo, -oo)
                         return S.false
-                    return S.true
+                    if lhs.is_extended_real and rhs.is_extended_real:
+                        L, R = [i.is_extended_positive for i in (lhs, rhs)]
+                        if L == R and L != None:
+                            return S.true
```

Comment:
According to codecov the lines inside this if aren't covered by the tests. Are they still needed?

Reply:
Not if I relax the condition for lhs/rhs being numbers in the above code.